oktapam/resource_security_policy.go

```diff
@@ -845,3 +878,33 @@ func readPrincipalsFromResourceData(d *schema.ResourceData) (*client.SecurityPol
 
 	return principals, diags
 }
+
+func resourceSecurityPolicyCustomizeDiff(ctx context.Context, diff *schema.ResourceDiff, v interface{}) error {
+	return validatePrincipalAccounts(diff)
+}
+
+func validatePrincipalAccounts(diff *schema.ResourceDiff) error {
```

Comment:
is the issue that you're trying to solve that the API won't accept a privilege that is not enabled but also has `admin_level_privileges` turned on?  or that it accepts it, but when the API returns the value, it doesn't return the `admin_level_privileges` value unless it is enabled?

Reply:
So, API will not accept a privilege that is not enabled but has `admin_level_privileges` enabled. I added a check in Terraform,  so that we don't even make an API call if it is an invalid request.